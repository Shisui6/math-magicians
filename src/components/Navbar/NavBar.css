nav {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 50px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  height: 72px;
}

nav ul {
  display: flex;
  list-style: none;
  padding: 0;
}

nav li {
  margin-left: 20px;
}

nav li:first-child {
  margin-left: 0;
}

nav h2 {
  color: #073066;
}

.link {
  text-decoration: none;
  color: #000;
  transition: 0.3s;
}

.link:hover {
  color: #073066;
}

.active {
  color: #073066;
  font-weight: 900;
  text-decoration: none;
}

@media screen and (max-width: 768px) {
  nav {
    height: 95px;
  }

  nav ul {
    flex-direction: column;
    align-items: flex-start;
  }

  nav li:first-child {
    margin-left: 20px;
  }
}
