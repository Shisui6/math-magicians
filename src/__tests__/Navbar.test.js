import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar/Navbar';

describe('Navbar', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Router>
        <Navbar />
      </Router>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the calculator link', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );

    expect(screen.getByText('Calculator')).toHaveClass('link');
  });
});
