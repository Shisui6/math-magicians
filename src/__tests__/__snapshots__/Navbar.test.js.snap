// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav>
  <h2>
    Math Magicians
  </h2>
  <div>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className="link"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
